tuningFork.c

```diff
@@ -203,93 +186,61 @@ void genFile(double freq, double duration, char* fname)
                 /* 32 */ 0x04, 0x00, 0x10, 0x00, 0x00, 0x00, 0x64, 0x61,
                 /* 40 */ 0x74, 0x61,   B4,   B3,   B2,   B1};
 
+    /////////////////////////
+    // Generate data subchunk
+    /////////////////////////
 
-    //////////////////////
-    // Calculate samples
-    //////////////////////
-
-
-    const int NUM_PERIODS = 400;
-    int16_t sample; // designates value of wave func at that point
     int max_volume = 0x7FFF;
-    double omega = 2*PI*freq; // cuts down on calculation time
-
-    int i;
-
-    double period = 1/freq; // in seconds
-    int tableSize = NUM_PERIODS * SAMPLE_RATE * period; // in samples
-    uint16_t* sampleTable = malloc(sizeof(uint16_t) * tableSize);
-
-    int dataLoopLimit = SAMPLE_RATE * duration;
-    uint16_t* dataArr = malloc(sizeof(uint16_t) * 2 * dataLoopLimit);
+    int numSamples = SAMPLE_RATE * duration;
 
     // Generate a lookup table for samples
-    //int nThreads = omp_get_num_procs();
-    //omp_set_num_threads(nThreads);
-
-    double omegaT = 0; // omega * time
-    double deltaOmegaT = (double)omega/SAMPLE_RATE;
-
-
-    for (i = 0; i < tableSize; i++)
-    {
-        //fprintf(stderr, "omegaT is %lf\n", omegaT);
-        sample = max_volume * sin(omegaT);
-        //sample = max_volume * sin(omega * (double)i/SAMPLE_RATE);
-
-        sampleTable[i] = sample;
-
-        omegaT += deltaOmegaT; // gives next value to evaluate sin at
+    const int lookupTableSize = 8192;  // Size in samples
```

Comment:
8192 is an arbitrary size that I chose. A full period of 8192 has a frequency of about 5Hz at a sample rate of 44100, and 5Hz is much lower than the range of human hearing. This should reduce the effect of aliasing. You can set it higher or lower if you want, though I would not set it lower than 2048, or about 20Hz (within human hearing range).